### PR TITLE
Exposing GetMostRecentDomains, modifying getRecentSubdomainsById

### PR DESCRIPTION
**Pull request by Will-Hyde**
Exposing GetMostRecentDomains so that it can be publicly accessible in the zns SDK.
Additionally, modifying getRecentSubdomainsbyId so that it fits more in line with GetMostRecentDomains (Updating name, and specifying a count to retrieve)

Waiting for the .env file to run the mocha tests I added and test it.

### CODE REVIEW

File: src/subgraph/actions/getMostRecentSubdomainsById.ts [hunk 1]
```diff
@@ -6,25 +6,30 @@ import { DomainsQueryDto } from "../types";
...
+export const getMostRecentSubdomainsById = async <T>(
   apolloClient: ApolloClient<T>,
-  domainId: string
+  domainId: string,
+  count = 1000
 ): Promise<Domain[]> => {
-  const queryCount = 1000;
   let skip = 0;
-
   const domains: Domain[] = [];
```
Comment (JamesEarle) [inline src/subgraph/actions/getMostRecentSubdomainsById.ts:17]: nit: consider renaming to `subdomains`

File: test/domains.test.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  
+    const subgraphClient = subgraph.createClient(config.subgraphUri);
+  
+    const domainIdToDomainName = async (domainId: string) => {
+      const domainData = await subgraphClient.getDomainById(domainId);
+      return domainData.name;
+    };
+
+    describe("Domain actions", () => {
+        it("gets 10 most recent domains", async () => {
```
Comment (JamesEarle) [inline test/domains.test.ts:58]: Add a test that fails as well, like requesting more than `5000`

**Review by JamesEarle — APPROVED** [review]

File: src/subgraph/actions/getMostRecentSubdomainsById.ts (hunk 1 above)
Comment (Will-Hyde) [reply src/subgraph/actions/getMostRecentSubdomainsById.ts:17]: makes sense to me, added

File: test/domains.test.ts (hunk 2 above)
Comment (Will-Hyde) [reply test/domains.test.ts:58]: Added

File: src/subgraph/actions/getMostRecentSubdomainsById.ts [hunk 3]
```diff
@@ -6,44 +6,50 @@ import { DomainsQueryDto } from "../types";
...
 
 const logger = getLogger().withTag("subgraph:actions:getRecentSubdomainsById");
+const MAX_RECORDS = 5000;
 
-export const getRecentSubdomainsById = async <T>(
+export const getMostRecentSubdomainsById = async <T>(
   apolloClient: ApolloClient<T>,
-  domainId: string
+  domainId: string,
+  count = 1000
```
Comment (RyanCross) [inline src/subgraph/actions/getMostRecentSubdomainsById.ts:14]: Consider letting the caller pass the `skip` starting value, then default to 0 like you are if not provided

File: src/subgraph/actions/getMostRecentSubdomainsById.ts [hunk 4]
```diff
@@ -6,44 +6,50 @@ import { DomainsQueryDto } from "../types";
...
 ): Promise<Domain[]> => {
-  const queryCount = 1000;
   let skip = 0;
```
Comment (RyanCross) [inline src/subgraph/actions/getMostRecentSubdomainsById.ts:16]: Consider letting the caller pass the skip starting value, and default to 0 if no arg is provided

File: src/subgraph/client.ts [hunk 5]
```diff
@@ -12,7 +12,7 @@ export interface SubgraphClient {
   getDomainsByName(name: string): Promise<Domain[]>;
   getDomainsByOwner(owner: string): Promise<Domain[]>;
   getSubdomainsById(domainId: string): Promise<Domain[]>;
-  getRecentSubdomainsById(domainId: string): Promise<Domain[]>;
+  getMostRecentSubdomainsById(domainId: string, count: number | undefined): Promise<Domain[]>;
```
Comment (RyanCross) [inline src/subgraph/client.ts:15]: A really handy shorthand for something like `count: number | undefined` is to add a `Maybe` type. 

The declaration looks like this:
`type Maybe<T> = T | undefined`

then you can just do `count: Maybe<number>`

**Review by Remscar — APPROVED** [review]

**Review by RyanCross — APPROVED** [review]

File: src/subgraph/actions/getMostRecentSubdomainsById.ts [hunk 6]
```diff
@@ -6,44 +6,50 @@ import { DomainsQueryDto } from "../types";
...
     }
 
     /**
      * We will only get back up to `queryCount` # of domains
      * So if we get that many there's probably more domains we need
      * to fetch. If we got back less, we can stop querying
      */
-    if (queriedDomains.length < queryCount) {
+     yetUnreceived -= queriedDomains.length;
+    if (queriedDomains.length < count || yetUnreceived <= 0) {
```
[the remark this answers is not in the document]
Comment (Will-Hyde) [reply src/subgraph/actions/getMostRecentSubdomainsById.ts:46]: I didn't actually write this logic, its actually from the GetMostRecentDomains script, but applied here to keep it unified in format.

If you like I can remove it for both scripts and open up a new review

**Pull request merged**